src/main/webapp/courseLoader.js

```diff
@@ -122,14 +125,21 @@
       prevTerm2 = `Fall ${String(parseInt(currentTerm[1] - 1))}`;
     }
 
-    const prevTermData1 = await getPreviousTermData(prevTerm1);
-    const prevTermData2 = await getPreviousTermData(prevTerm2);
+    let prevTermData1;
+    let prevTermData2;
+    await Promise.all([
```

Comment:
Is this actually doing what you want? Interesting if so.

Typically await and .then() are used separately.

For example, here I'd probably do something like:
```
const [prevTermData1, prevTermData2] = await Promise.all([getPreviousTermData(prevTerm1), getPreviousTermData(prevTerm2)]);
```
(This is using the array destructuring shorthand to make things cleaner)

This current implementation might end up working because the .then waits the same amount of time here as the await, so the variables will be set. But typically the .then() would happen after the rest of the function executed. So it's weird to have it in this format.

Reply:
Yah I think I need to read up on the await, promise, and async stuff. I feel like there is a decent gap in my knowledge on this topic.